src/test/java/teammates/test/driver/HtmlHelper.java

```diff
@@ -387,6 +388,15 @@ private static String replaceUnpredictableValuesWithPlaceholders(String content)
                                   + " value=\"" + REGEX_ENCRYPTED_REGKEY + "\"){3}",
                                   " name=\"" + Const.ParamsNames.REGKEY + "\""
                                   + " type=\"hidden\" value=\"\\${regkey\\.enc}\"")
+                      // anonymous student identifier on results page
+                      .replaceAll("Anonymous student " + REGEX_ANONYMOUS_PARTICIPANT_HASH,
+                                  "Anonymous student \\${student\\.hash}")
+                      // anonymous instructor identifier on results page
+                      .replaceAll("Anonymous instructor " + REGEX_ANONYMOUS_PARTICIPANT_HASH,
+                                  "Anonymous instructor \\${instructor\\.hash}")
+                      // anonymous team identifier on results page
+                      .replaceAll("Anonymous team " + REGEX_ANONYMOUS_PARTICIPANT_HASH,
+                                  "Anonymous team \\${team\\.hash}")
```

Comment:
If the hashes are all the same, I don't think we need three separate replaces. `.replaceAll("Anonymous (student|instructor|team) " + REGEX, "Anonymous $1 \\${participant\\.hash}")` should do the trick.